webhooks/views.py

```diff
@@ -0,0 +1,36 @@
+from django.http import HttpResponse
+from django.views.decorators.csrf import csrf_exempt
+
+from interface.models import Repo
+
+from webhooks.models import GitHook, GitHookError, GitHookAccessError
+
+
+@csrf_exempt
+def github_webhook(request):
+    git_hook = GitHook(request)
+
+    try:
+        git_hook.process_request()
+    except GitHookAccessError:
+        return HttpResponse(status=403)
+    except GitHookError as e:
+        return HttpResponse(str(e), status=400)
+
+    repo = Repo.objects.filter(full_name=git_hook.repository['full_name']).first()
```

Comment:
Not asking you to reconsider, but it's a good time to show you how the core Django team writes code.

It's interesting to look at the source code for `.get()`:
https://github.com/django/django/blob/master/django/db/models/query.py#L363

Usually, in the case of the object not existing, you'd return a 404. Django actually has a shortcut function for this, because its such a common action:
https://docs.djangoproject.com/en/1.11/topics/http/shortcuts/#get-object-or-404

But take a look at the source code of that shortcut:
https://github.com/django/django/blob/master/django/shortcuts.py#L75

The most common code path for fetching a single object or returning a 404 relies on an exception. Why? Because in Python, if the `.get()` is usually successful, then a 404 is truly exceptional, and it's easier to read the code if you see "Get or Does not exist"

As for exceptions as a part of normal workflows, in Python, it really matters which case is more likely. If the exception is less likely, then it's actually more efficient to use an exception than check the existence of a value. See this proof:
http://paltman.com/try-except-performance-in-python-a-simple-test/